File: Sources/Client/SortDirection.ts
/**
 * Specifies the direction of a sorted property.
 */
export const SortDirection = Object.freeze({

	/**
	 * The sort is ascending.
	 */
	Ascending: "Ascending",

	/**
	 * The sort is descending.
	 */
	Descending: "Descending"
});

/**
 * Specifies the direction of a sorted property.
 */
export type SortDirection = typeof SortDirection[keyof typeof SortDirection];

File: Sources/Server/SortDirection.cs
namespace Belin.UI;

using System.Text.Json.Serialization;

/// <summary>
/// Specifies the direction of a sorted property.
/// </summary>
[JsonConverter(typeof(JsonStringEnumConverter<SortDirection>))]
public enum SortDirection {

	/// <summary>
	/// The sort is ascending.
	/// </summary>
	Ascending,

	/// <summary>
	/// The sort is descending.
	/// </summary>
	Descending
}
